src/crafting.cpp

```diff
@@ -1746,219 +1791,277 @@ void set_item_inventory(item &newit)
     }
 }
 
+/* selection of component if a recipe requirement has multiple options (e.g. 'duct tap' or 'welder') */
+comp_selection player::select_component(const std::vector<item_comp> &components, int batch, bool can_cancel) {
+	std::vector<item_comp> player_has;
+	std::vector<item_comp> map_has;
+	std::vector<item_comp> mixed;
+
+	comp_selection selected = comp_selection();
```

Comment:
> Isn't it safer to initialize them?

That's why you have a constructor. This code:
```C++
comp_selection selected;
```
*automatically* calls the default constructor. You can see this all around the code, even two lines above, you see `std::vector<item_comp> mixed;`, which no explicit initialization. That seems to be fine, why does it bother you here?